### PR TITLE
add more info on converting xib programmatically

### PR DESCRIPTION
**Pull request by harrieshin**
want to add more info how to convert xib to programmatically defining uiview hierarchy in UIViewController

### CODE REVIEW

**Review by cy255 — APPROVED** [review]

File: StoryboardsAndXibs.md [hunk 1]
```diff
@@ -13,4 +13,15 @@ In most cases on both macOS and iOS, it is recommended to avoid storyboards (and
...
\ No newline at end of file
+- The main menu bar for a macOS app is required to be specified in a XIB.
+
+## How to convert xib to programmatic initialization
+- Change the UIViewController caller
+    - Instead of passing in [nibName](https://developer.apple.com/documentation/uikit/uiviewcontroller/1621359-init) in particular bundle, pass in nil
+    - Make sure no other storyboard is initializing with [iniWithCoder:](https://developer.apple.com/documentation/foundation/nscoding/1416145-initwithcoder?language=objc)
+- Initialize the UIViewController's view appropriately
+    - Override [loadView](https://developer.apple.com/documentation/uikit/uiviewcontroller/1621454-loadview?language=objc). If UIViewController needs to have a customer view, make sure you set the view in loadView. If there were anything that was declared in [awakeFromNib](https://developer.apple.com/documentation/objectivec/nsobject/1402907-awakefromnib?language=objc) make sure it has been moved over to loadView appropriately. For more info, please refer to [apple documentation](https://developer.apple.com/documentation/uikit/view_controllers/displaying_and_managing_views_with_a_view_controller?language=objc)
```
Comment (anandrajeswaran) [inline StoryboardsAndXibs.md:23]: "customer view" -> "custom view"

Comment (anandrajeswaran) [inline StoryboardsAndXibs.md:23]: "there were anything that was declared" -> "there was anything declared"

Comment (anandrajeswaran) [inline StoryboardsAndXibs.md:23]: For "apple documentation", capitalize "Apple"

File: StoryboardsAndXibs.md [hunk 2]
```diff
@@ -13,4 +13,15 @@ In most cases on both macOS and iOS, it is recommended to avoid storyboards (and
...
+    - Any properties that were declared with [IBOutlet](https://developer.apple.com/library/archive/documentation/General/Conceptual/CocoaEncyclopedia/Outlets/Outlets.html) now needs to be programmatically allocated it memory
```
Comment (anandrajeswaran) [inline StoryboardsAndXibs.md:24]: "now needs" -> "now need"

Comment (anandrajeswaran) [inline StoryboardsAndXibs.md:24]: "allocated it memory" -> "allocated in memory", but still that makes it sound much more complex than it needs to be? Should just be "can just be simply instantiated in loadView and added to the hierarchy"?

File: StoryboardsAndXibs.md [hunk 3]
```diff
@@ -13,4 +13,15 @@ In most cases on both macOS and iOS, it is recommended to avoid storyboards (and
...
+    - add any UIView to its self view's [hierarchy](https://developer.apple.com/documentation/uikit/uiview/1622616-addsubview?language=objc_).
```
Comment (anandrajeswaran) [inline StoryboardsAndXibs.md:25]: Isn't this kinda duplicate of the step above? Sample code may help here

File: StoryboardsAndXibs.md [hunk 4]
```diff
@@ -13,4 +13,15 @@ In most cases on both macOS and iOS, it is recommended to avoid storyboards (and
...
+- Initialize the UIViewController's view appropriately
+    - Override [loadView](https://developer.apple.com/documentation/uikit/uiviewcontroller/1621454-loadview?language=objc). If UIViewController needs to have a customer view, make sure you set the view in loadView. If there were anything that was declared in [awakeFromNib](https://developer.apple.com/documentation/objectivec/nsobject/1402907-awakefromnib?language=objc) make sure it has been moved over to loadView appropriately. For more info, please refer to [apple documentation](https://developer.apple.com/documentation/uikit/view_controllers/displaying_and_managing_views_with_a_view_controller?language=objc)
+    - Any properties that were declared with [IBOutlet](https://developer.apple.com/library/archive/documentation/General/Conceptual/CocoaEncyclopedia/Outlets/Outlets.html) now needs to be programmatically allocated it memory
+    - add any UIView to its self view's [hierarchy](https://developer.apple.com/documentation/uikit/uiview/1622616-addsubview?language=objc_). 
+    - set up all the subviews [constraints](https://developer.apple.com/documentation/uikit/nslayoutconstraint/1526955-activateconstraints/)
```
Comment (anandrajeswaran) [inline StoryboardsAndXibs.md:26]: "subviews constraints" -> "subviews' constraints. Probably should note in here to use NSStackView/UIStackView to minimize number of manual constraints needed?

File: StoryboardsAndXibs.md [hunk 5]
```diff
@@ -13,4 +13,15 @@ In most cases on both macOS and iOS, it is recommended to avoid storyboards (and
 
 ## Exceptions
 - Launch Screens on iOS are required to be specified as storyboards.
-- The main menu bar for a macOS app is required to be specified in a XIB.
\ No newline at end of file
+- The main menu bar for a macOS app is required to be specified in a XIB.
+
+## How to convert xib to programmatic initialization
+- Change the UIViewController caller
```
Comment (anandrajeswaran) [inline StoryboardsAndXibs.md:19]: Instructions here are entirely iOS and don't mention macOS at all?

**Pull request merged**